prog/weaveutil/container.go

```diff
@@ -0,0 +1,273 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+func containerID(args []string) error {
+	if len(args) < 1 {
+		cmdUsage("container-id", "<container-name-or-short-id>")
+	}
+	containerID := args[0]
+
+	c, err := docker.NewVersionedClientFromEnv("1.18")
+	if err != nil {
+		return fmt.Errorf("unable to connect to docker: %s", err)
+	}
+
+	container, err := c.InspectContainer(containerID)
+	if err != nil {
+		return fmt.Errorf("unable to inspect container %s: %s", containerID, err)
+	}
+	fmt.Print(container.ID)
+	return nil
+}
+
+func containerState(args []string) error {
+	if len(args) < 1 {
+		cmdUsage("container-state", "<container-id> [<image-name-or-id>]")
+	}
+	containerID := args[0]
+
+	c, err := docker.NewVersionedClientFromEnv("1.18")
+	if err != nil {
+		return fmt.Errorf("unable to connect to docker: %s", err)
+	}
+
+	container, err := c.InspectContainer(containerID)
+	if err != nil {
+		return fmt.Errorf("unable to inspect container %s: %s", containerID, err)
+	}
+
+	if len(args) == 1 {
+		fmt.Print(container.State.StateString())
+	} else {
+		image := args[1]
+		match1, _ := regexp.MatchString(image, container.Image)
```

Comment:
Why is regexp match the right thing to do here?

Reply:
`Image` is the `sha256:...` hash and `Config.Image` is the <name>:<tag>.  I was trying to support friendly name or hash for running the image comparison.    I guess for the `Image` field, I could just parse out the hash, but what if short hash was passed in.  As for `Config.Image`, I could parse out just the name without the tag (split on :) for the comparison.  The previous code did _not_ compare the tags, so I didn't want to change that.

What's your thoughts?  I was just trying to make the container-state check fairly generic.

Reply:
Changed to `strings.HasPrefix`